stl/inc/memory

```diff
@@ -3505,6 +3538,17 @@ _NODISCARD bool operator<=(nullptr_t _Left, const unique_ptr<_Ty, _Dx>& _Right)
     return !(_Right < _Left);
 }
 
+#ifdef __cpp_lib_concepts
+// clang-format off
+template <class _Ty, class _Dx>
+    requires three_way_comparable<typename unique_ptr<_Ty, _Dx>::pointer>
+_NODISCARD compare_three_way_result_t<typename unique_ptr<_Ty, _Dx>::pointer> operator<=>(
+    const unique_ptr<_Ty, _Dx>& _Left, nullptr_t) {
+    // clang-format on
+    return _Left.get() <=> static_cast<typename unique_ptr<_Ty, _Dx>::pointer>(nullptr);

```

Comment:
(Somehow I lost my original comment here :scream_cat: - rewriting it to explain the commit)

I think that this can also directly use `<=>` instead of `compare_three_way`; unlike the code above which uses `less<_Ptr>` exactly as depicted in the Standard, `compare_three_way` is not customizable and will always use `<=>` here (again with possible trickery for `constexpr` pointers which is inapplicable here). Using `<=>` directly will make `unique_ptr <=> nullptr_t` consistent with `unique_ptr <=> unique_ptr` being added above.